parts/k8s/kubernetesmastercustomscript.sh

```diff
@@ -566,35 +536,63 @@ function ensureJournal(){
     fi
 }
 
-function ensureApiserver() {
+function ensureK8s() {
     if $REBOOTREQUIRED; then
         return
     fi
-    kubernetesStarted=1
+    k8sHealthy=1
+    nodesActive=1
+    nodesReady=1
     for i in {1..600}; do
         if [ -e $KUBECTL ]
         then
-            $KUBECTL cluster-info
+            break
+        fi
+        sleep 1
+    done
+    for i in {1..600}; do
+        $KUBECTL 2>/dev/null cluster-info
             if [ "$?" = "0" ]
             then
-                echo "kubernetes started, took $i seconds"
-                kubernetesStarted=0
+                echo "k8s cluster is healthy, took $i seconds"
+                k8sHealthy=0
                 break
             fi
-        else
-            /usr/bin/docker ps | grep apiserver
-            if [ "$?" = "0" ]
+        sleep 1
+    done
+    if [ $k8sHealthy -ne 0 ]
+    then
+        echo "k8s cluster is not healthy after $i seconds"
+        exit 3
+    fi
+    for i in {1..1800}; do
+        nodes=$(${KUBECTL} get nodes 2>/dev/null | grep 'Ready' | wc -l)
```

Comment:
Since this is grepping for "Ready", is it possible that `nodesActive` would be set to 0 but that in the next step `nodesReady` wouldn't be 0?

Reply:
Yes, because `grep 'Ready'` will match both `Ready` and `NotReady`.

The purpose of `nodesActive` is to tag when all nodes have checked in with the api server (either `Ready` or `NotReady`). The purpose of `nodesReady` is to tag when all nodes are ready to receive workloads (i.e., `Ready`).